warehouse/ingest-core/src/main/java/datawave/ingest/table/aggregator/GlobalIndexUidAggregator.java

```diff
@@ -172,48 +169,49 @@ public Value reduce(Key key, Iterator<Value> iter) {
                     for (String uid : v.getUIDList()) {
                         
                         // check that a removal has not occurred
-                        // if it has, we decrement the count, from above.
                         if (!uidsToRemove.contains(uid) && !quarantinedIds.contains(uid)) {
                             
                             // add the UID iff we are under our MAX
                             if (uids.size() < maxUids)
                                 uids.add(uid);
```

Comment:
could add a break here if condition is false since we won't be adding any entries in the for loop once maxUids is reached.